guard/src/commands/parse_tree.rs

```diff
@@ -21,51 +21,50 @@ impl Command for ParseTree {
         PARSE_TREE
     }
 
-    fn command(&self) -> App<'static> {
-        App::new(PARSE_TREE)
-            .about(
-                r#"Prints out the parse tree for the rules defined in the file.
-"#,
-            )
+    fn command(&self) -> clap::Command {
+        clap::Command::new(PARSE_TREE)
+            .about("Prints out the parse tree for the rules defined in the file.")
             .arg(
-                Arg::with_name(RULES.0)
+                Arg::new(RULES.0)
                     .long(RULES.0)
                     .short(RULES.1)
-                    .takes_value(true)
-                    .help("Provid a rules file")
+                    .help("Provide a rules file")
+                    .action(ArgAction::Set)
                     .required(false),
             )
             .arg(
-                Arg::with_name(OUTPUT.0)
+                Arg::new(OUTPUT.0)
                     .long(OUTPUT.0)
                     .short(OUTPUT.1)
-                    .takes_value(true)
                     .help("Write to output file")
+                    .action(ArgAction::Set)
                     .required(false),
             )
             .arg(
-                Arg::with_name(PRINT_JSON.0)
+                Arg::new(PRINT_JSON.0)
                     .long(PRINT_JSON.0)
                     .short(PRINT_JSON.1)
-                    .required(false)
+                    .action(ArgAction::SetTrue)
                     .help("Print output in JSON format"),
             )
             .arg(
-                Arg::with_name(PRINT_YAML.0)
+                Arg::new(PRINT_YAML.0)
                     .long(PRINT_YAML.0)
                     .short(PRINT_YAML.1)
+                    .action(ArgAction::SetTrue)
                     .required(false)
                     .help("Print output in YAML format"),
             )
+            .arg_required_else_help(true)
     }
 
     fn execute(&self, app: &ArgMatches, writer: &mut Writer, reader: &mut Reader) -> Result<i32> {
-        let mut file: Box<dyn std::io::Read> = match app.value_of(RULES.0) {
+        let mut file: Box<dyn std::io::Read> = match app.get_one::<String>(RULES.0) {
             Some(file) => Box::new(std::io::BufReader::new(File::open(file)?)),
             None => Box::new(reader),
         };
 
-        let yaml = !app.is_present(PRINT_JSON.0);
+        let yaml = !app.get_flag(PRINT_JSON.0);
```

Comment:
Wait, we do have a `print-yaml` flag but we aren't using that? We are negating `print-json` istead?

Reply:
Yeah seems like thats the way they do it. Would you rather I just remove the flag alltogether ?